lib/sablon/image.rb

```diff
@@ -0,0 +1,17 @@
+module Sablon
+  class Image
+    include Singleton
+    attr_reader :definitions
+
+    Definition = Struct.new(:name, :data, :rid) do
+      def inspect
+        "#<Image #{name}:#{data}:#{rid}"
+      end
+    end
+
+    def self.create_by_path(path, random = nil)
```

Comment:
I'm not quite happy with the name `create_by_path`. It feels a bit verbose. What do you think about:

```ruby
def self.open(path, sablon_name: nil)
```